app/src/main/java/com/android/bookswap/model/map/BookManagerViewModel.kt

```diff
@@ -97,26 +97,34 @@ class BookManagerViewModel(
     var successUsers = false
     var currentAttempt = 0
     while ((!successBooks || !successUsers) && currentAttempt < MAXIMUM_RETRIES) {
-      userRepository.getUsers { users ->
-        if (users.isSuccess) {
-          _allUsers.value = users.getOrNull()!!
-          successUsers = true
-        } else {
-          Log.e("BookManagerViewModel", "Failed to fetch users.")
+      val userJob = coroutineScope {
```

Comment:
Same as before, while this is a good change, usually it's best to keep it in a separate PR.